### PR TITLE
Add D100C12S2 sku to SN4600C

### PR DESCRIPTION
**Pull request by alexrallen**
#### Why I did it

New SKU by customer request. 

```
4   100G Downlinks
2   10G  Downlinks
100 50G  Downlinks (2x Split)
8   100G Uplinks
```

#### How I did it

Duplicated existing 4600C SKU and updated
- Port Speeds
- Buffer Default Values
- Default Breakout Modes
- SAI Profile Speeds & Breakout Capabilities

#### How to verify it

Tested on 4600C with hardware loopback connections. 

#### Which release branch to backport (provide reason below if selected)

For the purposes of internal review this is based against master. A PR will be submitted individually against 202012 and 202106 / master.

#### Description for the changelog
[mellanox] Add D100C12S2 SKU to 4600C

#### A picture of a cute animal (not mandatory but encouraged)
alpaca-1

### CODE REVIEW

**Comment by stephenxs** [comment]
The numbers are correct.
Better to update the description to:
```
4  100G Downlinks
2  10G  Downlinks
100 50G  Downlinks
8  100G Uplinks
```
pg_profile_lookup.ini should be a symbol link to the same file in Mellanox-SN4600C-D112C8

**Comment by alexrallen** [comment]
> 
> 
> The numbers are correct.
> Better to update the description to:
> 
> ```
> 4  100G Downlinks
> 2  10G  Downlinks
> 100 50G  Downlinks
> 8  100G Uplinks
> ```
> 
> pg_profile_lookup.ini should be a symbol link to the same file in Mellanox-SN4600C-D112C8

This has been resolved by the latest commit.

**Review by stephenxs — APPROVED** [review]
Approved for buffer configurations.
Please check with other reviewers as well.

File: device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-D100C12S2/port_config.ini [hunk 1]
```diff
@@ -0,0 +1,114 @@
+Ethernet0      0,1                etp1a     1        50000     none
```
Comment (dgsudharsan) [inline device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-D100C12S2/port_config.ini:1]: Please add header. Please also remove 'fec'

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-D100C12S2/sai.profile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+SAI_INIT_CONFIG_FILE=/usr/share/sonic/hwsku/sai_4600c_100x50g_12x100g_2x10g.xml
+SAI_DUMP_STORE_PATH=/var/log/mellanox/sdk-dumps
+SAI_DUMP_STORE_AMOUNT=10
```
Comment (vivekreddynv) [inline device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-D100C12S2/sai.profile:3]: Is there any req regarding VxLAN source port range? If yes this should be set `SAI_VXLAN_SRCPORT_RANGE_ENABLE=1`

File: device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-D100C12S2/hwsku.json [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+            "default_brkout_mode": "2x50G[25G,10G,1G]"
+        },
+        "Ethernet252": {
+            "default_brkout_mode": "2x50G[25G,10G,1G]"
+        },
+        "Ethernet254": {
+            "default_brkout_mode": "2x50G[25G,10G,1G]"
+        }
+    }
+}
```
Comment (vivekreddynv) [inline device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-D100C12S2/hwsku.json:346]: newline terminator

File: device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-D100C12S2/port_config.ini [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+Ethernet232    464,465,466,467    etp59     59       100000    none
+Ethernet236    472,473,474,475    etp60     60       100000    none
+Ethernet240    480,481            etp61a    61       50000     none
+Ethernet242    482,483            etp61b    61       50000     none
+Ethernet244    488,489            etp62a    62       50000     none
+Ethernet246    490,491            etp62b    62       50000     none
+Ethernet248    496,497            etp63a    63       50000     none
+Ethernet250    498,499            etp63b    63       50000     none
+Ethernet252    504,505            etp64a    64       50000     none
+Ethernet254    506,507            etp64b    64       50000     none
```
Comment (vivekreddynv) [inline device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-D100C12S2/port_config.ini:114]: please add newline terminator

**Comment by alexrallen** [comment]
@dgsudharsan @vivekreddynv Updated per feedback.

**Review by vivekreddynv — APPROVED** [review]

File: device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-D100C12S2/hwsku.json [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+            "default_brkout_mode": "1x10G[100G,50G,40G,25G,1G]"
+        },
+        "Ethernet152": {
+            "default_brkout_mode": "2x50G[25G,10G,1G]"
+        },
+        "Ethernet154": {
+            "default_brkout_mode": "2x50G[25G,10G,1G]"
+        },
+        "Ethernet156": {
+            "default_brkout_mode": "1x10G[100G,50G,40G,25G,1G]"
```
Comment (dgsudharsan) [inline device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-D100C12S2/hwsku.json:223]: Shouldn't this mode be updated in platform.json too?

Comment (alexrallen) [reply device/mellanox/x86_64-mlnx_msn4600c-r0/Mellanox-SN4600C-D100C12S2/hwsku.json:223]: Yes. I hadn't included this in the PR but I will.

**Review by stephenxs — APPROVED** [review]
Approved for buffer with 2km supported.

**Review by vivekreddynv — APPROVED** [review]

**Review by dgsudharsan — APPROVED** [review]

**Pull request closed without merging**